### PR TITLE
Modified LOQ's detector to be more generic and work with SANS2D. Added more records to be checked by SANS2D

### PR DESCRIPTION
**Pull request by Adam-Szw**
### Description of work

https://github.com/ISISComputingGroup/IBEX/issues/4608
Ticket 4608

Implements shutter logic

### To test

*Which ticket does this PR fix?*

### Acceptance criteria

*List the acceptance criteria for the PR*

if the maximum safe whole detector or spectrum count rates are exceeded, the fast shutter is closed.

#### Code Review

- [ ] A copy of the manual has been placed on the shared drive
- [ ] Pertitent information has been stored in the [wiki](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Specific-Device-IOC)
- [ ] Does the IOC conform to IBEX standards?
    - [PV naming](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/PV-Naming).
    - [Disable records](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Disable-records)
    - [Record simulation](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Record-Simulation)
    - [Finishing touches](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/IOC-Finishing-Touches)
- [ ] If an OPI has been modified, does it conform to the [style guidelines](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/OPI-Creation)?
- [ ] Do the IOC and support module conform to the new [build guidelines](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Reducing-Build-Dependencies)
- [ ] Have the changes been recorded appropriately in a PR for [release notes](https://github.com/ISISComputingGroup/IBEX/blob/master/release_notes/ReleaseNotes_Upcoming.md)?
- [ ] Is the device's flow control setting correct? [For most devices flow control should be OFF](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Flow-control).

### Functional Tests

- IOC responds correctly in:
    - [ ] Devsim mode
    - [ ] Recsim mode
    - [ ] Real device, if available
- [ ] Supplementary IOCs (`..._0n` where `n>1`) run correctly
- [ ] Log files do not report undefined macros (serach for `macLib: macro` to find instances of `macLib: macro [macro name] is undefined...`

### Final steps

- [ ] Update the IOC submodule in the main EPICS repo. See [Git workflow](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Git-workflow) page for details.
- [ ] Reviewer has merged the associated PR for the [release notes](https://github.com/ISISComputingGroup/IBEX/blob/master/release_notes/ReleaseNotes_Upcoming.md)

### CODE REVIEW

**Review by JamesKingWork — CHANGES_REQUESTED** [review]

File: RUNCTRL/iocBoot/iocRUNCTRL_01/st.cmd [hunk 1]
```diff
@@ -33,12 +33,22 @@ dbLoadRecords("$(TOP)/db/runcontrolMgr.db","P=$(MYPVPREFIX),ALERT_OUT=$(MYPVPREF
...
 
-## load LOQ specific detector control, this puts in the aperture if the detctor count rate is exceeded
-stringiftest("LOQ", "$(ICPCONFIGHOST)", 5, "NDXLOQ")
-$(IFLOQ) dbLoadRecords("$(TOP)/db/LOQ_detector.db","P=$(MYPVPREFIX)")
-$(IFLOQ) dbLoadRecords("$(WEBGET)/db/sendAlert.db","P=$(MYPVPREFIX),Q=CS:DC:ALERTS:,INST=$(INSTRUMENT=Unknown),SOURCE=IBEX")
-$(IFLOQ) dbLoadRecords("$(RUNCONTROL)/db/gencontrolMgr.db","P=$(MYPVPREFIX),MODE=DC,OUT_ACTION=$(MYPVPREFIX)CS:OVERCOUNT:ALERT.PROC")
-$(IFLOQ) dbLoadRecords("$(RUNCONTROL)/db/gencontrol.db","P=$(MYPVPREFIX),MODE=DC,PV=$(MYPVPREFIX)DAE:AD1:INTG:SPEC:RATE,NOALIAS=#")
+## load detector control, this puts in the aperture if the detctor count rate is exceeded
+stringiftest("DETECT", "$(ICPCONFIGHOST)", 5, "NDXLOQ")
+stringiftest("DETECT", "$(ICPCONFIGHOST)", 5, "NDW1798")
```
[the remark this answers is not in the document]
Comment (Adam-Szw) [reply RUNCTRL/iocBoot/iocRUNCTRL_01/st.cmd:38]: 38 and 39, yes. Corrected now

File: RUNCTRL/iocBoot/iocRUNCTRL_01/st.cmd [hunk 2]
```diff
@@ -33,12 +33,31 @@ dbLoadRecords("$(TOP)/db/runcontrolMgr.db","P=$(MYPVPREFIX),ALERT_OUT=$(MYPVPREF
...
 iocshLoad "${ICPCONFIGROOT}/rc_settings.cmd", "RUNCONTROL=$(TOP)"
 
-## load LOQ specific detector control, this puts in the aperture if the detctor count rate is exceeded
-stringiftest("LOQ", "$(ICPCONFIGHOST)", 5, "NDXLOQ")
-$(IFLOQ) dbLoadRecords("$(TOP)/db/LOQ_detector.db","P=$(MYPVPREFIX)")
-$(IFLOQ) dbLoadRecords("$(WEBGET)/db/sendAlert.db","P=$(MYPVPREFIX),Q=CS:DC:ALERTS:,INST=$(INSTRUMENT=Unknown),SOURCE=IBEX")
-$(IFLOQ) dbLoadRecords("$(RUNCONTROL)/db/gencontrolMgr.db","P=$(MYPVPREFIX),MODE=DC,OUT_ACTION=$(MYPVPREFIX)CS:OVERCOUNT:ALERT.PROC")
-$(IFLOQ) dbLoadRecords("$(RUNCONTROL)/db/gencontrol.db","P=$(MYPVPREFIX),MODE=DC,PV=$(MYPVPREFIX)DAE:AD1:INTG:SPEC:RATE,NOALIAS=#")
+## load detector control, this puts in the aperture if the detctor count rate is exceeded
+stringiftest("DETECT", "$(ICPCONFIGHOST)", 5, "NDXLOQ")
```
Comment (daryakoskeroglu) [inline RUNCTRL/iocBoot/iocRUNCTRL_01/st.cmd:37]: Everything looks good to me, just a typo here `...aperture if the detector`

**Review by daryakoskeroglu — CHANGES_REQUESTED** [review]
Just a little typo, otherwise everything is ok

File: RUNCTRL/iocBoot/iocRUNCTRL_01/st.cmd (hunk 2 above)
Comment (Adam-Szw) [reply RUNCTRL/iocBoot/iocRUNCTRL_01/st.cmd:37]: This was updated to be more descriptive

**Review by daryakoskeroglu — APPROVED** [review]

**Pull request merged**